src/shared/modules/discovery/discoveryDuck.js

```diff
@@ -79,17 +79,21 @@ export const getBoltHost = state => {
 }
 
 const updateDiscoveryState = (action, store) => {
-  const updateObj = { host: action.forceURL }
-  if (action.username) {
-    updateObj.username = action.username
-  }
-  if (action.password) {
-    updateObj.password = action.password
-  }
+  const keysToCopy = [
+    'username',
+    'password',
+    'connectTo',
+    'restApi',
+    'showDbField'
+  ]
+  const updateObj = keysToCopy.reduce(
+    (accObj, key) => (action[key] ? { ...accObj, [key]: action[key] } : accObj),
+    { host: action.forceURL }
+  )
+
   if (typeof action.encrypted !== 'undefined') {
```

Comment:
Is there a reason this was checked to be undefined rather than truthyness? @oskarhane